src/SFA.DAS.EAS.Portal/Application/Commands/Cohort/CohortApprovalRequestedCommandHandler.cs

```diff
@@ -5,35 +5,54 @@
 using System;
 using SFA.DAS.EAS.Portal.Application.Services;
 using SFA.DAS.EAS.Portal.Client.Types;
+using SFA.DAS.EAS.Portal.Application.Commands.Account;
+using SFA.DAS.HashingService;
 
 namespace SFA.DAS.EAS.Portal.Application.Commands.Cohort
 {
     public class CohortApprovalRequestedCommandHandler : ICommandHandler<CohortApprovalRequestedCommand>
     {
         private readonly IAccountDocumentService _accountsService;
         private readonly IProviderCommitmentsApi _providerCommitmentsApi;
+        private readonly ICommandHandler<AccountCreatedCommand> _accountCreatedHandler;
+        private readonly IHashingService _hashingService;
 
         public CohortApprovalRequestedCommandHandler(
             IAccountDocumentService accountsService, 
-            IProviderCommitmentsApi providerCommitmentsApi)
+            IProviderCommitmentsApi providerCommitmentsApi,
+            ICommandHandler<AccountCreatedCommand> accountCreatedHandler,
+            IHashingService hashingService)
         {
             _accountsService = accountsService;
             _providerCommitmentsApi = providerCommitmentsApi;
+            _accountCreatedHandler = accountCreatedHandler;
+            _hashingService = hashingService;
         }
 
         public async Task Handle(CohortApprovalRequestedCommand command, CancellationToken cancellationToken = default)
         {
-            var accountDocumentTask = _accountsService.Get(command.AccountId, cancellationToken);
-            var commitmentTask = _providerCommitmentsApi.GetProviderCommitment(command.ProviderId, command.CommitmentId);
-            
-            var accountDocument = await accountDocumentTask;
+            var commitment = await _providerCommitmentsApi.GetProviderCommitment(command.ProviderId, command.CommitmentId);            
+            long accountLegalEntityId = _hashingService.DecodeValue(commitment.AccountLegalEntityPublicHashedId);
+
+            await _accountCreatedHandler.Handle(new AccountCreatedCommand(command.AccountId, commitment.LegalEntityName), cancellationToken);
```

Comment:
I think we should consider simplifying account creation. With the introduction of the `AccountCreatedCommand`,
* we've lost the concurrency in `CohortApprovalRequestedCommandHandler`, when calling the commitments API and getting the account from our store
* we now retrieve the account twice in both the reservation and cohort command handlers
* we can now end up saving the account twice in the reservation and cohort command handlers

The extra complexity
* makes the code more difficult to maintain
* makes it harder to think through how the system works
* introduces more run-time processing and memory churn, slowing down the time it takes to process events

Instead of introducing a new Command and Command handler, we could just add a new helper method somewhere (so we maintain single responsibility), that essentially just encapsulates these lines (from the original reservation command)...

```
           var accountDocument = await _accountsService.Get(reservedFunding.AccountId, cancellationToken);

            if (accountDocument == null)
            {
                accountDocument = AccountDocument.Create(reservedFunding.AccountId);
```
So that we have access to the context of whether the account had to be created or not, it'd need to return both the account and whether it's had to be created (we could return a tuple or use an out parameter).